In [23]:
import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn import svm
from sklearn.metrics import accuracy_score,classification_report
from sklearn.preprocessing import StandardScaler
from sklearn.tree import DecisionTreeClassifier
from sklearn.linear_model import LogisticRegression

In [24]:
data = pd.read_csv("parkinsons.csv")
data.head()

name  MDVP:Fo(Hz)  MDVP:Fhi(Hz)  ...   spread2        D2       PPE
0  phon_R01_S01_1      119.992       157.302  ...  0.266482  2.301442  0.284654
1  phon_R01_S01_2      122.400       148.650  ...  0.335590  2.486855  0.368674
2  phon_R01_S01_3      116.682       131.111  ...  0.311173  2.342259  0.332634
3  phon_R01_S01_4      116.676       137.871  ...  0.334147  2.405554  0.368975
4  phon_R01_S01_5      116.014       141.781  ...  0.234513  2.332180  0.410335

[5 rows x 24 columns]

In [25]:
parkinsons_data = data.drop(columns=["name"],axis=1)

In [26]:
parkinsons_data.shape

(195, 23)

In [27]:
parkinsons_data.describe()

MDVP:Fo(Hz)  MDVP:Fhi(Hz)  ...          D2         PPE
count   195.000000    195.000000  ...  195.000000  195.000000
mean    154.228641    197.104918  ...    2.381826    0.206552
std      41.390065     91.491548  ...    0.382799    0.090119
min      88.333000    102.145000  ...    1.423287    0.044539
25%     117.572000    134.862500  ...    2.099125    0.137451
50%     148.790000    175.829000  ...    2.361532    0.194052
75%     182.769000    224.205500  ...    2.636456    0.252980
max     260.105000    592.030000  ...    3.671155    0.527367

[8 rows x 23 columns]

In [28]:
parkinsons_data["status"].value_counts()

status
1    147
0     48
Name: count, dtype: int64

In [29]:
parkinsons_data.groupby("status").mean()

MDVP:Fo(Hz)  MDVP:Fhi(Hz)  MDVP:Flo(Hz)  ...   spread2        D2       PPE
status                                           ...                              
0        181.937771    223.636750    145.207292  ...  0.160292  2.154491  0.123017
1        145.180762    188.441463    106.893558  ...  0.248133  2.456058  0.233828

[2 rows x 22 columns]

In [30]:
x = parkinsons_data.drop(columns=["status"],axis=1)
y = parkinsons_data["status"]
x.head()

MDVP:Fo(Hz)  MDVP:Fhi(Hz)  MDVP:Flo(Hz)  ...   spread2        D2       PPE
0      119.992       157.302        74.997  ...  0.266482  2.301442  0.284654
1      122.400       148.650       113.819  ...  0.335590  2.486855  0.368674
2      116.682       131.111       111.555  ...  0.311173  2.342259  0.332634
3      116.676       137.871       111.366  ...  0.334147  2.405554  0.368975
4      116.014       141.781       110.655  ...  0.234513  2.332180  0.410335

[5 rows x 22 columns]

In [31]:
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size=0.2,random_state=42)
x_train.head()

MDVP:Fo(Hz)  MDVP:Fhi(Hz)  MDVP:Flo(Hz)  ...   spread2        D2       PPE
5        120.552       131.162       113.787  ...  0.299111  2.187560  0.357775
135      110.453       127.611       105.554  ...  0.322044  2.631793  0.228319
122      138.190       203.522        83.340  ...  0.160267  2.090438  0.229892
167      260.105       264.919       237.303  ...  0.160414  1.881767  0.075587
85       180.978       200.125       155.495  ...  0.315903  3.098256  0.200423

[5 rows x 22 columns]

In [32]:
x_test.shape

(39, 22)

In [33]:
svm_model=svm.SVC(kernel="linear")
svm_model.fit(x_train,y_train)

SVC(kernel='linear')

In [34]:
svm_pred=svm_model.predict(x_test)
print(accuracy_score(y_test,svm_pred))
print( classification_report(y_test, svm_pred))

0.8717948717948718
              precision    recall  f1-score   support

           0       0.75      0.43      0.55         7
           1       0.89      0.97      0.93        32

    accuracy                           0.87        39
   macro avg       0.82      0.70      0.74        39
weighted avg       0.86      0.87      0.86        39



In [35]:

#Decision tree 
dt_model = DecisionTreeClassifier(random_state=42)
dt_model.fit(x_train, y_train)
dt_pred = dt_model.predict(x_test)

print(accuracy_score(y_test, dt_pred))
print(classification_report(y_test, dt_pred))

0.9230769230769231
              precision    recall  f1-score   support

           0       0.83      0.71      0.77         7
           1       0.94      0.97      0.95        32

    accuracy                           0.92        39
   macro avg       0.89      0.84      0.86        39
weighted avg       0.92      0.92      0.92        39



In [36]:
#Logistic regression
lr_model = LogisticRegression()
lr_model.fit(x_train, y_train)
lr_pred = lr_model.predict(x_test)

print("Logistic Regression Accuracy:", accuracy_score(y_test, lr_pred))
print("Logistic Regression Report:\n", classification_report(y_test, lr_pred))

Logistic Regression Accuracy: 0.8974358974358975
Logistic Regression Report:
               precision    recall  f1-score   support

           0       1.00      0.43      0.60         7
           1       0.89      1.00      0.94        32

    accuracy                           0.90        39
   macro avg       0.94      0.71      0.77        39
weighted avg       0.91      0.90      0.88        39



C:\Users\prata\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\LocalCache\local-packages\Python312\site-packages\sklearn\linear_model\_logistic.py:465: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. OF ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


Choosing decision tree model since its accuracy is 92.3%

In [38]:
import pickle
filename = "parkinsons.sav"
pickle.dump(dt_model,open(filename,'wb'))